feat(locale): add location for uz_UZ_latin

Mirazyzz opened the pull request:
<!-- Please run `pnpm run preflight` before opening a Pull Request to ensure that your code fulfills the minimal requirements for our project. -->

<!-- Please first read https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md -->

<!-- Help us by writing a correct PR title following this guide: https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md#committing -->

Mirazyzz committed:
add location locale

Mirazyzz committed:
remove duplicates

ST-DDT commented on src/locales/uz_UZ_latin/location/street_pattern.ts:
```diff
@@ -0,0 +1,4 @@
+export default [
+  '{{location.street_suffix}} {{location.street_name}}',
```
This looks strange to me.
Why does it start with the suffix?
Also IMO we should name the street_name file street_name_part or similar.

Mirazyzz replied:
I missed that one, suffix should be after the street name.

ST-DDT reviewed (commented):
Please sort the files alphabetically.

Mirazyzz commented:
> Please sort the files alphabetically.

Do you want the street names to be sorted alphabetically? They're pretty large and looking at other locales I thought it's not necessary.

Shinigami92 commented:
I suggest to add the locale to the PR title

Shinigami92 reviewed (dismissed):
Count me already as approved but fix the change request from @ST-DDT pls

matthewmayer commented:
> > Please sort the files alphabetically.
> 
> 
> 
> Do you want the street names to be sorted alphabetically? They're pretty large and looking at other locales I thought it's not necessary.

All *new* locale data should be sorted alphabetically unless there's a good reason (eg months of the year can be kept as Jan-Feb-mar...)

ST-DDT reviewed (changes requested):
Please make the requested changes.

Mirazyzz committed:
PR review fixes

Fix incorrect country name, use proper prefix for street pattern, add additional secondary pattern

Mirazyzz committed:
rename street_name to street_name_part

Mirazyzz commented:
> Please make the requested changes.

Ready! sorry for the delay.

matthewmayer commented on src/locales/uz_UZ_latin/location/street_pattern.ts:
```diff
@@ -0,0 +1 @@
+export default ['{{location.street_name}} {{location.street_suffix}}'];
```
This should be changed to location.street_name_part now right?

Mirazyzz replied:
exactly, my bad.

Mirazyzz committed:
Update street_pattern.ts

ST-DDT committed:
Merge branch 'next' into extend-locale-location

matthewmayer commented on src/locales/uz_UZ_latin/location/street_address.ts:
```diff
@@ -0,0 +1,4 @@
+export default {
+  normal: '{{location.street}}, {{location.buildingNumber}}',
+  full: '{{location.street}}, {{location.buildingNumber}} {{location.secondaryAddress}}',
```
Is this correct?
If a secondary address (e.g. room/apartment number is provided) this immediately follows the building number with no comma etc?

without secondary address:
```
fakerUZ_UZ_latin.location.streetAddress()
> Munif Mahanov qishlog'i, 905
> Sobit Miranov xiyoboni, 35a
```

with secondary address:
```
fakerUZ_UZ_latin.location.streetAddress(true)
> Munif Mahanov qishlog'i, 905 972 uy
> Sobit Miranov xiyoboni, 35a 817 xonadon
```

Mirazyzz replied:
Yes, you're right I think it's better to separate them with comma. By the way, it's also better to put "-" for secondary address between house number and title (uy, xonadon). In Uzbek "-" is used for ordinal numbers. Do you think I should change it too?

matthewmayer replied:
Yes

Mirazyzz committed:
separate address with comma & update secondary address

matthewmayer committed:
Merge branch 'next' into extend-locale-location
